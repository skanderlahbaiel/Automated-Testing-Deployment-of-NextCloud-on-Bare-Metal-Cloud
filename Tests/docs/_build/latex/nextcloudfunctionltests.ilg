This is makeindex, version 2.15 [TeX Live 2022/dev] (kpathsea + Thai support).
Scanning style file ./python.ist.......done (7 attributes redefined, 0 ignored).
Scanning input file nextcloudfunctionltests.idx....done (132 entries accepted, 0 rejected).
Sorting entries....done (1123 comparisons).
Generating output file nextcloudfunctionltests.ind....done (272 lines written, 0 warnings).
Output written in nextcloudfunctionltests.ind.
Transcript written in nextcloudfunctionltests.ilg.
