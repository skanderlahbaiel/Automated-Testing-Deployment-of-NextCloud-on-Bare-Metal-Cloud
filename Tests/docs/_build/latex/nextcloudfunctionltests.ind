\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\let\spxpagem \sphinxstyleindexpagemain
\let\spxentry \sphinxstyleindexentry
\let\spxextra \sphinxstyleindexextra

  \bigletter A
  \item \spxentry{add\_user()}\spxextra{in module tests.test\_create\_user}, \hyperpage{15}
  \item \spxentry{add\_user\_and\_check\_presence()}\spxextra{in module utils.user\_management}, 
		\hyperpage{31}

  \indexspace
  \bigletter C
  \item \spxentry{calculate\_sha256()}\spxextra{in module utils.file\_management}, \hyperpage{26}
  \item \spxentry{check\_external\_file\_sharing()}\spxextra{in module utils.file\_management}, 
		\hyperpage{26}
  \item \spxentry{check\_file\_deletion()}\spxextra{in module utils.file\_management}, 
		\hyperpage{26}
  \item \spxentry{check\_file\_presence()}\spxextra{in module utils.file\_management}, 
		\hyperpage{26}
  \item \spxentry{check\_files\_container\_presence()}\spxextra{in module utils.file\_management}, 
		\hyperpage{26}
  \item \spxentry{check\_if\_loggedIn()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{24}
  \item \spxentry{check\_internal\_file\_share()}\spxextra{in module utils.file\_management}, 
		\hyperpage{27}
  \item \spxentry{check\_internal\_sharing()}\spxextra{in module utils.file\_management}, 
		\hyperpage{27}
  \item \spxentry{check\_logout()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{24}
  \item \spxentry{check\_presence\_user\_management\_page()}\spxextra{in module utils.user\_management}, 
		\hyperpage{32}
  \item \spxentry{check\_successful\_page\_load()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{24}
  \item \spxentry{check\_url\_login\_page()}\spxextra{in module utils.file\_management}, 
		\hyperpage{27}
  \item \spxentry{check\_user\_absence()}\spxextra{in module utils.user\_management}, 
		\hyperpage{32}
  \item \spxentry{check\_user\_presence()}\spxextra{in module utils.user\_management}, 
		\hyperpage{32}
  \item \spxentry{cleanup}
    \subitem \spxentry{module}, \hyperpage{3}
  \item \spxentry{cleanup.delete\_test\_files}
    \subitem \spxentry{module}, \hyperpage{1}
  \item \spxentry{click\_new\_fileOrFolder()}\spxextra{in module utils.file\_management}, 
		\hyperpage{27}
  \item \spxentry{click\_on\_create\_user()}\spxextra{in module utils.user\_management}, 
		\hyperpage{33}
  \item \spxentry{click\_on\_logout()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{25}
  \item \spxentry{click\_users\_button()}\spxextra{in module utils.user\_management}, 
		\hyperpage{33}
  \item \spxentry{compare\_hashes()}\spxextra{in module utils.file\_management}, \hyperpage{27}
  \item \spxentry{config}
    \subitem \spxentry{module}, \hyperpage{5}
  \item \spxentry{config.configuration}
    \subitem \spxentry{module}, \hyperpage{3}
  \item \spxentry{create\_file()}\spxextra{in module utils.file\_management}, \hyperpage{27}
  \item \spxentry{create\_testing\_files()}\spxextra{in module utils.file\_management}, 
		\hyperpage{27}

  \indexspace
  \bigletter D
  \item \spxentry{delete\_file()}\spxextra{in module tests.test\_file\_delete}, \hyperpage{17}
  \item \spxentry{delete\_files\_starting\_with()}\spxextra{in module cleanup.delete\_test\_files}, 
		\hyperpage{1}
  \item \spxentry{delete\_user()}\spxextra{in module tests.test\_delete\_user}, \hyperpage{16}
  \item \spxentry{download\_file()}\spxextra{in module tests.test\_file\_download}, \hyperpage{17}
  \item \spxentry{driver()}\spxextra{in module file\_management\_cycle}, \hyperpage{6}
  \item \spxentry{driver()}\spxextra{in module utils.webdriver\_setup}, \hyperpage{36}

  \indexspace
  \bigletter E
  \item \spxentry{ensure\_element\_interactable()}\spxextra{in module utils.file\_management}, 
		\hyperpage{27}
  \item \spxentry{ensure\_logged\_in\_and\_goto\_files()}\spxextra{in module utils.file\_management}, 
		\hyperpage{27}
  \item \spxentry{extract\_file\_type()}\spxextra{in module utils.file\_management}, \hyperpage{28}
  \item \spxentry{extract\_filename()}\spxextra{in module utils.file\_management}, \hyperpage{28}

  \indexspace
  \bigletter F
  \item \spxentry{file\_integrity()}\spxextra{in module tests.test\_file\_integrity}, 
		\hyperpage{18}
  \item \spxentry{file\_management\_cycle}
    \subitem \spxentry{module}, \hyperpage{5}
  \item \spxentry{fill\_user\_form()}\spxextra{in module utils.user\_management}, \hyperpage{33}
  \item \spxentry{find\_delete\_button()}\spxextra{in module utils.file\_management}, 
		\hyperpage{28}

  \indexspace
  \bigletter G
  \item \spxentry{generate\_confirm\_delete\_user\_button\_xpath()}\spxextra{in module utils.user\_management}, 
		\hyperpage{34}
  \item \spxentry{generate\_delete\_user\_button\_xpath()}\spxextra{in module utils.user\_management}, 
		\hyperpage{34}
  \item \spxentry{generate\_email()}\spxextra{in module utils.user\_management}, \hyperpage{34}
  \item \spxentry{generate\_external\_share\_verification\_element()}\spxextra{in module utils.file\_management}, 
		\hyperpage{28}
  \item \spxentry{generate\_file\_share\_button()}\spxextra{in module utils.file\_management}, 
		\hyperpage{28}
  \item \spxentry{generate\_internal\_sharing\_verification\_element\_exists()}\spxextra{in module utils.file\_management}, 
		\hyperpage{29}
  \item \spxentry{generate\_name()}\spxextra{in module utils.user\_management}, \hyperpage{34}
  \item \spxentry{generate\_password()}\spxextra{in module utils.user\_management}, \hyperpage{34}
  \item \spxentry{generate\_user()}\spxextra{in module utils.user\_management}, \hyperpage{34}
  \item \spxentry{generate\_user\_actions\_button\_xpath()}\spxextra{in module utils.user\_management}, 
		\hyperpage{35}
  \item \spxentry{generate\_user\_xpath()}\spxextra{in module utils.user\_management}, 
		\hyperpage{35}
  \item \spxentry{get\_external\_share\_link()}\spxextra{in module utils.file\_management}, 
		\hyperpage{29}
  \item \spxentry{get\_internal\_share\_link()}\spxextra{in module utils.file\_management}, 
		\hyperpage{29}
  \item \spxentry{get\_token\_from\_cookies()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{25}
  \item \spxentry{goto\_files()}\spxextra{in module utils.file\_management}, \hyperpage{29}
  \item \spxentry{green\_text()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{25}

  \indexspace
  \bigletter I
  \item \spxentry{ifLogin\_delete\_cookies()}\spxextra{in module utils.file\_management}, 
		\hyperpage{29}
  \item \spxentry{is\_element\_clickable()}\spxextra{in module utils.file\_management}, 
		\hyperpage{30}

  \indexspace
  \bigletter L
  \item \spxentry{load\_nextcloud\_page()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{25}
  \item \spxentry{locate\_and\_click\_share\_button()}\spxextra{in module tests.test\_file\_share}, 
		\hyperpage{19}
  \item \spxentry{locate\_file\_in\_nextcloud()}\spxextra{in module utils.file\_management}, 
		\hyperpage{30}
  \item \spxentry{login()}\spxextra{in module tests.test\_login}, \hyperpage{22}
  \item \spxentry{login\_to\_nextcloud()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{25}
  \item \spxentry{logout()}\spxextra{in module tests.test\_logout}, \hyperpage{23}
  \item \spxentry{logout\_and\_goto\_external\_link()}\spxextra{in module utils.file\_management}, 
		\hyperpage{30}

  \indexspace
  \bigletter M
  \item \spxentry{main()}\spxextra{in module cleanup.delete\_test\_files}, \hyperpage{2}
  \item \spxentry{modify\_quota()}\spxextra{in module utils.user\_management}, \hyperpage{35}
  \item \spxentry{modify\_user\_quota()}\spxextra{in module tests.test\_modify\_quota}, 
		\hyperpage{24}
  \item \spxentry{module}
    \subitem \spxentry{cleanup}, \hyperpage{3}
    \subitem \spxentry{cleanup.delete\_test\_files}, \hyperpage{1}
    \subitem \spxentry{config}, \hyperpage{5}
    \subitem \spxentry{config.configuration}, \hyperpage{3}
    \subitem \spxentry{file\_management\_cycle}, \hyperpage{5}
    \subitem \spxentry{tests}, \hyperpage{24}
    \subitem \spxentry{tests.end\_to\_end\_encryption}, \hyperpage{8}
    \subitem \spxentry{tests.test\_create\_user}, \hyperpage{15}
    \subitem \spxentry{tests.test\_delete\_user}, \hyperpage{16}
    \subitem \spxentry{tests.test\_file\_delete}, \hyperpage{17}
    \subitem \spxentry{tests.test\_file\_download}, \hyperpage{17}
    \subitem \spxentry{tests.test\_file\_integrity}, \hyperpage{18}
    \subitem \spxentry{tests.test\_file\_share}, \hyperpage{19}
    \subitem \spxentry{tests.test\_file\_upload}, \hyperpage{21}
    \subitem \spxentry{tests.test\_login}, \hyperpage{22}
    \subitem \spxentry{tests.test\_logout}, \hyperpage{23}
    \subitem \spxentry{tests.test\_modify\_quota}, \hyperpage{24}
    \subitem \spxentry{utils}, \hyperpage{36}
    \subitem \spxentry{utils.authentication\_management}, \hyperpage{24}
    \subitem \spxentry{utils.file\_creation}, \hyperpage{26}
    \subitem \spxentry{utils.file\_management}, \hyperpage{26}
    \subitem \spxentry{utils.user\_management}, \hyperpage{31}
    \subitem \spxentry{utils.webdriver\_setup}, \hyperpage{36}

  \indexspace
  \bigletter N
  \item \spxentry{navigate\_to\_users\_page()}\spxextra{in module utils.user\_management}, 
		\hyperpage{35}

  \indexspace
  \bigletter O
  \item \spxentry{open\_new\_user\_form()}\spxextra{in module utils.user\_management}, 
		\hyperpage{35}
  \item \spxentry{orange\_text()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{25}

  \indexspace
  \bigletter P
  \item \spxentry{perform\_delete\_file()}\spxextra{in module utils.file\_management}, 
		\hyperpage{30}
  \item \spxentry{perform\_delete\_user()}\spxextra{in module utils.user\_management}, 
		\hyperpage{35}
  \item \spxentry{perform\_download\_file()}\spxextra{in module utils.file\_management}, 
		\hyperpage{30}
  \item \spxentry{perform\_file\_upload()}\spxextra{in module utils.file\_management}, 
		\hyperpage{30}
  \item \spxentry{perform\_logout()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{25}

  \indexspace
  \bigletter R
  \item \spxentry{refresh\_page()}\spxextra{in module utils.user\_management}, \hyperpage{35}

  \indexspace
  \bigletter S
  \item \spxentry{scroll\_element\_into\_view()}\spxextra{in module utils.file\_management}, 
		\hyperpage{31}
  \item \spxentry{scroll\_to\_bottom\_dynamic()}\spxextra{in module utils.file\_management}, 
		\hyperpage{31}
  \item \spxentry{scroll\_with\_mouse\_wheel()}\spxextra{in module utils.file\_management}, 
		\hyperpage{31}
  \item \spxentry{share\_file()}\spxextra{in module tests.test\_file\_share}, \hyperpage{20}
  \item \spxentry{share\_file\_with\_recipient()}\spxextra{in module utils.file\_management}, 
		\hyperpage{31}

  \indexspace
  \bigletter T
  \item \spxentry{test\_file\_management\_cycle()}\spxextra{in module file\_management\_cycle}, 
		\hyperpage{6}
  \item \spxentry{test\_files()}\spxextra{in module file\_management\_cycle}, \hyperpage{7}
  \item \spxentry{test\_quota()}\spxextra{in module utils.user\_management}, \hyperpage{36}
  \item \spxentry{tests}
    \subitem \spxentry{module}, \hyperpage{24}
  \item \spxentry{tests.end\_to\_end\_encryption}
    \subitem \spxentry{module}, \hyperpage{8}
  \item \spxentry{tests.test\_create\_user}
    \subitem \spxentry{module}, \hyperpage{15}
  \item \spxentry{tests.test\_delete\_user}
    \subitem \spxentry{module}, \hyperpage{16}
  \item \spxentry{tests.test\_file\_delete}
    \subitem \spxentry{module}, \hyperpage{17}
  \item \spxentry{tests.test\_file\_download}
    \subitem \spxentry{module}, \hyperpage{17}
  \item \spxentry{tests.test\_file\_integrity}
    \subitem \spxentry{module}, \hyperpage{18}
  \item \spxentry{tests.test\_file\_share}
    \subitem \spxentry{module}, \hyperpage{19}
  \item \spxentry{tests.test\_file\_upload}
    \subitem \spxentry{module}, \hyperpage{21}
  \item \spxentry{tests.test\_login}
    \subitem \spxentry{module}, \hyperpage{22}
  \item \spxentry{tests.test\_logout}
    \subitem \spxentry{module}, \hyperpage{23}
  \item \spxentry{tests.test\_modify\_quota}
    \subitem \spxentry{module}, \hyperpage{24}
  \item \spxentry{toggle\_file\_upload()}\spxextra{in module utils.file\_management}, 
		\hyperpage{31}
  \item \spxentry{toggle\_settings\_menu()}\spxextra{in module utils.authentication\_management}, 
		\hyperpage{26}

  \indexspace
  \bigletter U
  \item \spxentry{upload\_file()}\spxextra{in module tests.test\_file\_upload}, \hyperpage{21}
  \item \spxentry{utils}
    \subitem \spxentry{module}, \hyperpage{36}
  \item \spxentry{utils.authentication\_management}
    \subitem \spxentry{module}, \hyperpage{24}
  \item \spxentry{utils.file\_creation}
    \subitem \spxentry{module}, \hyperpage{26}
  \item \spxentry{utils.file\_management}
    \subitem \spxentry{module}, \hyperpage{26}
  \item \spxentry{utils.user\_management}
    \subitem \spxentry{module}, \hyperpage{31}
  \item \spxentry{utils.webdriver\_setup}
    \subitem \spxentry{module}, \hyperpage{36}

  \indexspace
  \bigletter W
  \item \spxentry{wait\_then\_click()}\spxextra{in module utils.user\_management}, \hyperpage{36}
  \item \spxentry{wait\_then\_send\_keys()}\spxextra{in module utils.user\_management}, 
		\hyperpage{36}

\end{sphinxtheindex}
