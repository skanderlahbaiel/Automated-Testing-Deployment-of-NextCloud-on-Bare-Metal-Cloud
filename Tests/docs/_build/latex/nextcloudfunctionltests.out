\BOOKMARK [0][-]{chapter.1}{\376\377\000f\000u\000n\000c\000t\000i\000o\000n\000a\000l\000\137\000t\000e\000s\000t\000i\000n\000g}{}% 1
\BOOKMARK [1][-]{section.1.1}{\376\377\000c\000l\000e\000a\000n\000u\000p\000\040\000p\000a\000c\000k\000a\000g\000e}{chapter.1}% 2
\BOOKMARK [2][-]{subsection.1.1.1}{\376\377\000S\000u\000b\000m\000o\000d\000u\000l\000e\000s}{section.1.1}% 3
\BOOKMARK [2][-]{subsection.1.1.2}{\376\377\000c\000l\000e\000a\000n\000u\000p\000.\000d\000e\000l\000e\000t\000e\000\137\000t\000e\000s\000t\000\137\000f\000i\000l\000e\000s\000\040\000m\000o\000d\000u\000l\000e}{section.1.1}% 4
\BOOKMARK [3][-]{subsubsection*.4}{\376\377\000E\000n\000v\000i\000r\000o\000n\000m\000e\000n\000t\000\040\000V\000a\000r\000i\000a\000b\000l\000e\000s\000:}{subsection.1.1.2}% 5
\BOOKMARK [3][-]{subsubsection*.5}{\376\377\000S\000e\000t\000t\000i\000n\000g\000\040\000E\000n\000v\000i\000r\000o\000n\000m\000e\000n\000t\000\040\000V\000a\000r\000i\000a\000b\000l\000e\000s\000:}{subsection.1.1.2}% 6
\BOOKMARK [4][-]{paragraph*.6}{\376\377\000F\000o\000r\000\040\000U\000n\000i\000x\000/\000L\000i\000n\000u\000x\000/\000m\000a\000c\000O\000S\000:}{subsubsection*.5}% 7
\BOOKMARK [4][-]{paragraph*.7}{\376\377\000F\000o\000r\000\040\000W\000i\000n\000d\000o\000w\000s\000:}{subsubsection*.5}% 8
\BOOKMARK [3][-]{subsubsection*.8}{\376\377\000F\000u\000n\000c\000t\000i\000o\000n\000s\000:}{subsection.1.1.2}% 9
\BOOKMARK [3][-]{subsubsection*.9}{\376\377\000U\000s\000a\000g\000e\000:}{subsection.1.1.2}% 10
\BOOKMARK [2][-]{subsection.1.1.3}{\376\377\000M\000o\000d\000u\000l\000e\000\040\000c\000o\000n\000t\000e\000n\000t\000s}{section.1.1}% 11
\BOOKMARK [1][-]{section.1.2}{\376\377\000c\000o\000n\000f\000i\000g\000\040\000p\000a\000c\000k\000a\000g\000e}{chapter.1}% 12
\BOOKMARK [2][-]{subsection.1.2.1}{\376\377\000S\000u\000b\000m\000o\000d\000u\000l\000e\000s}{section.1.2}% 13
\BOOKMARK [2][-]{subsection.1.2.2}{\376\377\000c\000o\000n\000f\000i\000g\000.\000c\000o\000n\000f\000i\000g\000u\000r\000a\000t\000i\000o\000n\000\040\000m\000o\000d\000u\000l\000e}{section.1.2}% 14
\BOOKMARK [3][-]{subsubsection*.11}{\376\377\000S\000e\000t\000t\000i\000n\000g\000\040\000E\000n\000v\000i\000r\000o\000n\000m\000e\000n\000t\000\040\000V\000a\000r\000i\000a\000b\000l\000e\000s}{subsection.1.2.2}% 15
\BOOKMARK [4][-]{paragraph*.12}{\376\377\000U\000n\000i\000x\000-\000l\000i\000k\000e\000\040\000s\000y\000s\000t\000e\000m\000s\000\040\000\050\000L\000i\000n\000u\000x\000/\000m\000a\000c\000O\000S\000\051}{subsubsection*.11}% 16
\BOOKMARK [4][-]{paragraph*.13}{\376\377\000W\000i\000n\000d\000o\000w\000s\000\040\000s\000y\000s\000t\000e\000m\000s}{subsubsection*.11}% 17
\BOOKMARK [3][-]{subsubsection*.14}{\376\377\000U\000s\000a\000g\000e}{subsection.1.2.2}% 18
\BOOKMARK [3][-]{subsubsection*.15}{\376\377\000X\000P\000a\000t\000h\000\040\000L\000o\000c\000a\000t\000o\000r\000s}{subsection.1.2.2}% 19
\BOOKMARK [2][-]{subsection.1.2.3}{\376\377\000M\000o\000d\000u\000l\000e\000\040\000c\000o\000n\000t\000e\000n\000t\000s}{section.1.2}% 20
\BOOKMARK [1][-]{section.1.3}{\376\377\000f\000i\000l\000e\000\137\000m\000a\000n\000a\000g\000e\000m\000e\000n\000t\000\137\000c\000y\000c\000l\000e\000\040\000m\000o\000d\000u\000l\000e}{chapter.1}% 21
\BOOKMARK [2][-]{subsection.1.3.1}{\376\377\000A\000u\000t\000o\000m\000a\000t\000e\000d\000\040\000T\000e\000s\000t\000\040\000S\000e\000q\000u\000e\000n\000c\000e\000\040\000f\000o\000r\000\040\000F\000i\000l\000e\000\040\000a\000n\000d\000\040\000U\000s\000e\000r\000\040\000M\000a\000n\000a\000g\000e\000m\000e\000n\000t\000\040\000F\000u\000n\000c\000t\000i\000o\000n\000a\000l\000i\000t\000i\000e\000s}{section.1.3}% 22
\BOOKMARK [3][-]{subsubsection*.16}{\376\377\000E\000x\000e\000c\000u\000t\000i\000o\000n\000\040\000C\000o\000n\000t\000e\000x\000t}{subsection.1.3.1}% 23
\BOOKMARK [2][-]{subsection.1.3.2}{\376\377\000E\000x\000a\000m\000p\000l\000e\000\040\000U\000s\000a\000g\000e}{section.1.3}% 24
\BOOKMARK [2][-]{subsection.1.3.3}{\376\377\000W\000o\000r\000k\000f\000l\000o\000w}{section.1.3}% 25
\BOOKMARK [1][-]{section.1.4}{\376\377\000t\000e\000s\000t\000s\000\040\000p\000a\000c\000k\000a\000g\000e}{chapter.1}% 26
\BOOKMARK [2][-]{subsection.1.4.1}{\376\377\000S\000u\000b\000m\000o\000d\000u\000l\000e\000s}{section.1.4}% 27
\BOOKMARK [2][-]{subsection.1.4.2}{\376\377\000t\000e\000s\000t\000s\000.\000e\000n\000d\000\137\000t\000o\000\137\000e\000n\000d\000\137\000e\000n\000c\000r\000y\000p\000t\000i\000o\000n\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 28
\BOOKMARK [3][-]{subsubsection*.20}{\376\377\000T\000e\000s\000t\000i\000n\000g\000\040\000E\000n\000d\000-\000t\000o\000-\000E\000n\000d\000\040\000E\000n\000c\000r\000y\000p\000t\000i\000o\000n\000\040\000\050\000E\0002\000E\000E\000\051\000\040\000i\000n\000\040\000N\000e\000x\000t\000c\000l\000o\000u\000d\000\040\000C\000l\000i\000e\000n\000t}{subsection.1.4.2}% 29
\BOOKMARK [4][-]{paragraph*.21}{\376\377\000P\000r\000e\000r\000e\000q\000u\000i\000s\000i\000t\000e\000s}{subsubsection*.20}% 30
\BOOKMARK [4][-]{paragraph*.22}{\376\377\000S\000t\000e\000p\000s}{subsubsection*.20}% 31
\BOOKMARK [4][-]{paragraph*.23}{\376\377\000C\000o\000n\000c\000l\000u\000s\000i\000o\000n}{subsubsection*.20}% 32
\BOOKMARK [4][-]{paragraph*.24}{\376\377\000N\000o\000t\000e}{subsubsection*.20}% 33
\BOOKMARK [2][-]{subsection.1.4.3}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000c\000r\000e\000a\000t\000e\000\137\000u\000s\000e\000r\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 34
\BOOKMARK [2][-]{subsection.1.4.4}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000d\000e\000l\000e\000t\000e\000\137\000u\000s\000e\000r\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 35
\BOOKMARK [2][-]{subsection.1.4.5}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000f\000i\000l\000e\000\137\000d\000e\000l\000e\000t\000e\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 36
\BOOKMARK [2][-]{subsection.1.4.6}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000f\000i\000l\000e\000\137\000d\000o\000w\000n\000l\000o\000a\000d\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 37
\BOOKMARK [2][-]{subsection.1.4.7}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000f\000i\000l\000e\000\137\000i\000n\000t\000e\000g\000r\000i\000t\000y\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 38
\BOOKMARK [2][-]{subsection.1.4.8}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000f\000i\000l\000e\000\137\000s\000h\000a\000r\000e\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 39
\BOOKMARK [2][-]{subsection.1.4.9}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000f\000i\000l\000e\000\137\000u\000p\000l\000o\000a\000d\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 40
\BOOKMARK [2][-]{subsection.1.4.10}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000l\000o\000g\000i\000n\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 41
\BOOKMARK [2][-]{subsection.1.4.11}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000l\000o\000g\000o\000u\000t\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 42
\BOOKMARK [2][-]{subsection.1.4.12}{\376\377\000t\000e\000s\000t\000s\000.\000t\000e\000s\000t\000\137\000m\000o\000d\000i\000f\000y\000\137\000q\000u\000o\000t\000a\000\040\000m\000o\000d\000u\000l\000e}{section.1.4}% 43
\BOOKMARK [2][-]{subsection.1.4.13}{\376\377\000M\000o\000d\000u\000l\000e\000\040\000c\000o\000n\000t\000e\000n\000t\000s}{section.1.4}% 44
\BOOKMARK [1][-]{section.1.5}{\376\377\000u\000t\000i\000l\000s\000\040\000p\000a\000c\000k\000a\000g\000e}{chapter.1}% 45
\BOOKMARK [2][-]{subsection.1.5.1}{\376\377\000S\000u\000b\000m\000o\000d\000u\000l\000e\000s}{section.1.5}% 46
\BOOKMARK [2][-]{subsection.1.5.2}{\376\377\000u\000t\000i\000l\000s\000.\000a\000u\000t\000h\000e\000n\000t\000i\000c\000a\000t\000i\000o\000n\000\137\000m\000a\000n\000a\000g\000e\000m\000e\000n\000t\000\040\000m\000o\000d\000u\000l\000e}{section.1.5}% 47
\BOOKMARK [2][-]{subsection.1.5.3}{\376\377\000u\000t\000i\000l\000s\000.\000f\000i\000l\000e\000\137\000c\000r\000e\000a\000t\000i\000o\000n\000\040\000m\000o\000d\000u\000l\000e}{section.1.5}% 48
\BOOKMARK [2][-]{subsection.1.5.4}{\376\377\000u\000t\000i\000l\000s\000.\000f\000i\000l\000e\000\137\000m\000a\000n\000a\000g\000e\000m\000e\000n\000t\000\040\000m\000o\000d\000u\000l\000e}{section.1.5}% 49
\BOOKMARK [2][-]{subsection.1.5.5}{\376\377\000u\000t\000i\000l\000s\000.\000u\000s\000e\000r\000\137\000m\000a\000n\000a\000g\000e\000m\000e\000n\000t\000\040\000m\000o\000d\000u\000l\000e}{section.1.5}% 50
\BOOKMARK [2][-]{subsection.1.5.6}{\376\377\000u\000t\000i\000l\000s\000.\000w\000e\000b\000d\000r\000i\000v\000e\000r\000\137\000s\000e\000t\000u\000p\000\040\000m\000o\000d\000u\000l\000e}{section.1.5}% 51
\BOOKMARK [2][-]{subsection.1.5.7}{\376\377\000M\000o\000d\000u\000l\000e\000\040\000c\000o\000n\000t\000e\000n\000t\000s}{section.1.5}% 52
\BOOKMARK [0][-]{chapter.2}{\376\377\000I\000n\000d\000i\000c\000e\000s\000\040\000a\000n\000d\000\040\000t\000a\000b\000l\000e\000s}{}% 53
\BOOKMARK [0][-]{section*.106}{\376\377\000P\000y\000t\000h\000o\000n\000\040\000M\000o\000d\000u\000l\000e\000\040\000I\000n\000d\000e\000x}{}% 54
\BOOKMARK [0][-]{section*.107}{\376\377\000I\000n\000d\000e\000x}{}% 55
